app/workers/functions.php

```diff
@@ -1,100 +1,315 @@
 <?php
 
+require_once __DIR__ . '/../worker.php';
+
+use Utopia\Queue;
+use Utopia\Queue\Message;
 use Appwrite\Event\Event;
-use Appwrite\Event\Func;
 use Appwrite\Messaging\Adapter\Realtime;
-use Appwrite\Resque\Worker;
 use Appwrite\Usage\Stats;
 use Appwrite\Utopia\Response\Model\Execution;
-use Cron\CronExpression;
 use Executor\Executor;
 use Utopia\App;
 use Utopia\CLI\Console;
 use Utopia\Config\Config;
 use Utopia\Database\Database;
-use Utopia\Database\DateTime;
 use Utopia\Database\Document;
 use Utopia\Database\ID;
 use Utopia\Database\Permission;
 use Utopia\Database\Query;
 use Utopia\Database\Role;
+use Utopia\Database\Validator\Authorization;
+use Utopia\Logger\Log;
+
+Authorization::disable();
+Authorization::setDefaultStatus(false);
+
+global $client;
+global $workerNumber;
+
+$executor = new Executor(App::getEnv('_APP_EXECUTOR_HOST'));
+
+$execute = function (
+    Document $project,
+    Document $function,
+    Database $dbForProject,
+    string $trigger,
+    string $executionId = null,
+    string $event = null,
+    string $eventData = null,
+    string $data = null,
+    ?Document $user = null,
+    string $jwt = null
+) use ($executor) {
+
+    $user ??= new Document();
+    $functionId = $function->getId();
+    $deploymentId = $function->getAttribute('deployment', '');
+
+    /** Check if deployment exists */
+    $deployment = $dbForProject->getDocument('deployments', $deploymentId);
+
+    if ($deployment->getAttribute('resourceId') !== $functionId) {
+        throw new Exception('Deployment not found. Create deployment before trying to execute a function', 404);
```

Comment:
We're not in an HTTP server context these 404/500 error codes has no meaning.

Reply:
Error codes removed